packages/workspace/src/browser/workspace-uri-contribution.ts

```diff
@@ -73,8 +80,20 @@ export class WorkspaceUriLabelProviderContribution extends DefaultUriLabelProvid
         return icon;
     }
 
+    protected readonly onElementUpdatedEmitter = new Emitter<URI>();
+    protected testUris: Map<string, URI> = new Map();
```

Comment:
I wonder how in the real case one can get all these URIs? Won't it be potentially perf issue as well?

Reply:
> I wonder how in the real case one can get all these URIs? Won't it be potentially perf issue as well?

Actually even in this contrived test case there is no need to keep a map.  The `affects` function could have simply tested for 'test' in the URL.  In a real life case it would be even less likely to be a problem as the `affects` function needs only to know enough about the change so it can test if it affects a given uri/object.